hivemq-edge/src/main/java/com/hivemq/mqtt/services/PublishDistributorImpl.java

```diff
@@ -148,25 +189,58 @@ private ListenableFuture<PublishStatus> handlePublish(@NotNull final PUBLISH pub
             return Futures.immediateFuture(NOT_CONNECTED);
         }
 
-        return queuePublish(client, publish, subscriptionQos, false, retainAsPublished,
-                subscriptionIdentifier, clientSession.getQueueLimit());
+        return queuePublish(client,
+                publish,
+                subscriptionQos,
+                false,
+                retainAsPublished,
+                subscriptionIdentifier,
+                clientSession.getQueueLimit());
     }
 
     @NotNull
-    private SettableFuture<PublishStatus> queuePublish(@NotNull final String client, @NotNull final PUBLISH publish,
-                                                       final int subscriptionQos, final boolean shared, final boolean retainAsPublished,
-                                                       @Nullable final ImmutableIntArray subscriptionIdentifier,
-                                                       @Nullable final Long queueLimit) {
+    private SettableFuture<PublishStatus> queuePublish(
+            @NotNull final String client,
+            @NotNull final PUBLISH publish,
+            final int subscriptionQos,
+            final boolean shared,
+            final boolean retainAsPublished,
+            @Nullable final ImmutableIntArray subscriptionIdentifier,
+            @Nullable final Long queueLimit) {
+
+
+        Long appliedQueueLimit = queueLimit;
+        int appliedQoS = subscriptionQos;
+        if (shared) {
+            // update with the configuration of the bridge, if it is a bridge client
```

Comment:
check if id starts with forwarder then we don't need to iterate the bridges if it is not.

Could be placed in `handlePublish` there we already do the qos and queue size checks for normal subs

Reply:
yeah makes sense. I'll put it there.